src/core/PaperScript.js

```diff
@@ -195,20 +195,16 @@ Base.exports.PaperScript = function() {
                             + arg + ')');
                 }
                 break;
-            case 'BinaryExpression': // a + b, a - b, a / b, a * b, a == b, ...
+             case 'BinaryExpression': // a + b, a - b, a / b, a * b, a == b, ...
                 if (node.operator in binaryOperators
                         && node.left.type !== 'Literal') {
                     var left = getCode(node.left),
                         right = getCode(node.right),
                         between = getBetween(node.left, node.right),
                         operator = node.operator;
-                    replaceCode(node, '__$__(' + left + ','
-                            // To preserve line-breaks, get the code in between
-                            // left & right, and replace the occurrence of the
-                            // operator with its string counterpart:
-                            + between.replace(new RegExp('\\' + operator),
-                                '"' + operator + '"')
-                            + ', ' + right + ')');
+                        var string = 
+                             '__$__(' + left+ ',' + '"' + operator + '"' + ', ' + right + ')';
```

Comment:
One could restore line-number correctness by simply counting the number of linebreaks in `between` and adding them before or after the operator: `'\n'.repeat((between.match(/\n/g) || []).length)`